services/command-router/src/main/java/org/eclipse/hono/commandrouter/CommandRouterService.java

```diff
@@ -77,6 +79,7 @@ Future<CommandRouterResult> setLastKnownGatewayForDevice(String tenantId, Map<St
      *
      * @param tenantId The tenant id.
      * @param deviceId The device id.
+     * @param sendEvent {@code true} if <em>connected notification</em> event should be sent.
```

Comment:
is there a case where this will be `false`? my understanding of the discussion in #2760 is that we want to completely move the responsibility for sending the *connected*/*disconnected* events to the Command Router, or am I mistaken?

Reply:
As agreed with @calohmn this PR is about the MQTT adapter only. For the moment the other adapters will use `false`.
Later with other PRs we could adjust the others. But in general I think the direction is to move that responsibility to the Command Router for all adapters. As there is no difference between them on that.